src/org/parosproxy/paros/view/MainFrame.java

```diff
@@ -644,12 +647,37 @@ public void setShowTabNames(boolean showTabNames) {
 	/**
 	 * Sets the title of the main window.
 	 * <p>
+	 * The actual title set is the current session name, then the filename of the session (if persisted), 
+	 * followed by the program name and version.
+	 * 
+	 * @param session the {@code Session} from which the window title is being built
+	 * 
+	 * @see Constant#PROGRAM_NAME
+	 * @see Constant#PROGRAM_VERSION
+	 * @since TODO add version
+	 */
+	public void setTitle(Session session) {
+		StringBuilder strBuilder = new StringBuilder();
+		strBuilder.append(session.getSessionName()).append(" - ");

```

Comment:
Missed this one, sorry! We need to check that the session is not null (it might be null when discarding a session).